### ISSUE TITLE
Support creating client specific configs for different buckets

### ISSUE THREAD

**Issue by matthewmturner** [datafusion-contrib/datafusion-objectstore-s3#25]
Good work :+1:  Just a note for future improvements, it's very common to setup different access control for different buckets, so we will need to support creating different clients with specific configs for different buckets in the future. For example, in our production environment, we have spark jobs that access different buckets hosted in different AWS accounts.

_Originally posted by @houqp in https://github.com/datafusion-contrib/datafusion-objectstore-s3/issues/20#issuecomment-1019425889_

**Comment by matthewmturner** [comment]
@seddonm1 FYI created this to continue conversations on the topic.  

Do you think that this should be a requirement before publishing the crate?

**Comment by seddonm1** [comment]
@matthewmturner I think this is an edge-case but up to @houqp to answer.

Most users will never use this functionality so I think we can easily publish a 0.1 pending the DataFusion release then this can be added after.

**Comment by matthewmturner** [comment]
@seddonm1 i saw you raised https://github.com/awslabs/aws-sdk-rust/issues/425.  
Would you like something like what was proposed to be added as a type of credentials provider?

**Comment by houqp** [comment]
@seddonm1 definitely not a blocker for crates.io release :) Just a feature we can work on later.

**Comment by seddonm1** [comment]
@matthewmturner that request was around being able to access public buckets which is independent to this request

**Comment by matthewmturner** [comment]
@seddonm1 yes understood that its separate from this - just wasnt sure if you wanted to add a new issue for that functionality.
